tools/cldr-apps/js/src/esm/cldrSideways.mjs

```diff
@@ -121,115 +138,131 @@ function setMenuFromData(regionalVariantsWrapper, json, cacheKey) {
-          }
-        }
-
-        // replace the default locale(read-only) with base locale, store its name for label
-        for (let l = 0; l < list.length; l++) {
-          const loc = list[l][0];
-          const bund = locmap.getLocaleInfo(loc);
-          if (bund && bund.readonly) {
-            readLocale = locmap.getRegionAndOrVariantName(loc);
-            list[l][0] = topLocale;
-            list[l][1] = baseValue;
-            break;
-          }
-        }
-      };
-
-      // compare all sublocale values
-      function appendLocaleList(list, curValue) {
-        popupSelect.label = "Regional Variants for " + curLocaleName;
-
-        for (let l = 0; l < list.length; l++) {
-          const loc = list[l][0];
-          const title = list[l][1];
-          const item = { value: loc };
-          let str = locmap.getRegionAndOrVariantName(loc);
-          if (loc === topLocale) {
-            str += " (= " + readLocale + ")";
-          }
-
-          if (loc === cldrStatus.getCurrentLocale()) {
-            str = escape + str;
-            item.disabled = true;
-          } else if (title != curValue) {
-            str = unequalSign + str;
-            item.disabled = false;
-          } else {
-            str = escape + str;
-            item.disabled = false;
-          }
-          item.str = str;
-          popupSelect.items.push(item);
-        }
-      }
+      setMenuFromNontrivialData(regionalVariantsWrapper, json, cacheKey);
+    }
+  }
+}
 
-      const dataList = [];
-      const popupSelect = {};
-      popupSelect.items = [];
-      for (let s in json.others) {
-        for (let t in json.others[s]) {
-          dataList.push([json.others[s][t], s]);
-        }
-      }
+function setMenuFromNontrivialData(regionalVariantsWrapper, json, cacheKey) {
+  const dataList = initializeDataList(json.others);
+  const curValue = getCurrentValue(dataList);
 
-      /*
-       * Set curValue = the value for cldrStatus.getCurrentLocale()
-       */
-      let curValue = null;
-      for (let l = 0; l < dataList.length; l++) {
-        const loc = dataList[l][0];
-        if (loc === cldrStatus.getCurrentLocale()) {
-          curValue = dataList[l][1];
-          break;
-        }
-      }
-      /*
-       * Force the use of unequalSign in the regional comparison pop-up for locales in
-       * json.novalue, by assigning a value that's different from curValue.
-       */
-      if (json.novalue) {
-        const differentValue = curValue === "A" ? "B" : "A"; // anything different from curValue
-        for (let s in json.novalue) {
-          dataList.push([json.novalue[s], differentValue]);
-        }
-      }
-      mergeReadBase(dataList);
+  /*
+   * Force the use of unequalSign in the regional comparison pop-up for locales in
+   * json.novalue, by assigning a value that's different from curValue.
+   */
+  if (json.novalue) {
+    const differentValue = curValue === "A" ? "B" : "A"; // anything different from curValue
```

Comment:
```suggestion
    const differentValue = NaN;
```

It'll never be === any other string.   Or it could even be `const differentvalue = Symbol("no value")`;